server/core/region.go

```diff
@@ -88,9 +116,20 @@ func (r *RegionInfo) Clone() *RegionInfo {
 	for _, peer := range r.PendingPeers {
 		pendingPeers = append(pendingPeers, proto.Clone(peer).(*metapb.Peer))
 	}
+	learners := make([]*metapb.Peer, 0, len(r.Learners))
+	for _, peer := range r.Learners {
+		learners = append(learners, proto.Clone(peer).(*metapb.Peer))
+	}
+	voters := make([]*metapb.Peer, 0, len(r.Learners))
+	for _, peer := range r.Voters {
+		voters = append(voters, proto.Clone(peer).(*metapb.Peer))
+	}
```

Comment:
I'm concern about the memory allocate cost here.
Maybe we can set _Voters_ to _Region.Peers_ directly if there are no learner.

Reply:
or point to Region.Peers?